controllers/category.js

```diff
@@ -0,0 +1,244 @@
+const { upload } = require('../helpers/helper');
+
+/**
+ * @api {get} /category/:id Request Category 
+ * @apiName GetCategory
+ * @apiGroup Category
+ *
+ * @apiParam {Number} id Category unique ID.
+ *
+ * @apiSuccess {String} categoryName CategoryName of the product.
+ * @apiSuccess {String}  subCategories subCategories of the product.
+ * @apiSuccess {Onject} categoryImages categoryImages of the product.
+ * @apiSuccess {Number} price price of the product.
+ * 
+ * 
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "categoryName": "women Ethics",
+ *       "subCategories": "sarees"
+ *        "categoryImages":"productImages"
+ *        "price":"price"
+ *        
+ *     }
+ *
+ * @apiError CategoryNotFound The id of the User was not found.
+ *
+ * @apiErrorExample Error-Response:
+ *     HTTP/1.1 404 Not Found
+ *     {
+ *       "error": "CategoryNotFound"
+ *     }
+ * 
+ */
+
+exports.insertCategory = async (req, res, next) => {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      res.status(422).json({ errors: errors.array() });
+      return;
+    }
+    let imagesArray = [];
+    req.files.forEach(element => {
+      const file = {
+        imageName: element.originalname,
+        imagePath: element.path,
+        imageType: element.mimetype,
+        fileSize: fileSizeFormatter(element.size, 2)
+      }
+      imagesArray.push(file);
+    });
+    const categoryData = new Category({
+      categoryName: req.body.categoryName,
+      subCategories: req.body.subCategories,
+      categoryImages: imagesArray,
+      price: req.body.price,
+    });
+    await categoryData.save();
+    res.status(201).send({
+      status:true,
+      massage:'Category saved and images Uploaded Successfully',
+      data:categoryData,
+    });
+  } catch (error) {
+    res.status(400).send(error.message);
+  }
+}
+
+const fileSizeFormatter = (bytes, decimal) => {
+  if (bytes === 0) {
+    return '0 Bytes';
+  }
+  const dm = decimal || 2;
+  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'YB', 'ZB'];
+  const index = Math.floor(Math.log(bytes) / Math.log(1000));
+  return parseFloat((bytes / Math.pow(1000, index)).toFixed(dm)) + ' ' + sizes[index];
+
+}
+
+/**
+ * @api {put} /Category/ Modify Category information
+ * @apiName PutCategory
+ * @apiGroup Category
+ *
+ * @apiParam {Number} id  Category unique ID.
+ * @apiParam {String} [categoryName] categoryName of the product.
+ * @apiParam {String} [subategories]  subCategories of the product.
+ *
+ * @apiParam {Object} [categoryImages]  CategoryImages of the product.
+ *
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *
+ * @apiUse CategoryNotFoundError
+ */
+
+//update data
+exports.update = async (req, res) => {
+  try {
+    
+    const updatedCartegory = await Category.findByIdAndUpdate(
+      req.params.id,
+      {
+        $set: req.body,

```

Comment:
why update does not have logic for updating images?

Reply:
how can i update form data? updating only raw body data